public/app/features/alerting/unified/components/silences/SilencesDrawer.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { Drawer, Stack } from '@grafana/ui';
+import { AlertManagerPicker } from 'app/features/alerting/unified/components/AlertManagerPicker';
+import { GrafanaAlertmanagerDeliveryWarning } from 'app/features/alerting/unified/components/GrafanaAlertmanagerDeliveryWarning';
+import { SilencesEditor } from 'app/features/alerting/unified/components/silences/SilencesEditor';
+import { getDefaultSilenceFormValues } from 'app/features/alerting/unified/components/silences/utils';
+import { useAlertmanager } from 'app/features/alerting/unified/state/AlertmanagerContext';
+import { isGrafanaRulerRule } from 'app/features/alerting/unified/utils/rules';
+import { CombinedRule } from 'app/types/unified-alerting';
+
+type Props = {
+  rule: CombinedRule;
+  onClose: () => void;
+};
+
+/**
+ * For a given rule, renders a drawer containing silences editor and Alertmanager selection
+ */
+const SilencesDrawer = ({ rule, onClose }: Props) => {
+  const ruleUid = isGrafanaRulerRule(rule.rulerRule) ? rule.rulerRule?.grafana_alert.uid : '';
```

Comment:
What if instead of having `rule: CombinedRule` in props we would have `rule: RulerGrafanaRuleDTO`?
Is there a reason why we would like to show the drawer for non-Grafana rules? I feel it could be misleading because for Mimir/Loki rules the same UI (drawer) would result in a different behavior (creating a global silence instead of rule-level one)

Reply:
Yes definitely - I much prefer that. It makes it clearer that we only expect this drawer to be shown for GMA (for now, at least)